ihatemoney/web.py

```diff
@@ -654,12 +653,7 @@ def list_bills():
     # set the last selected payer as default choice if exists
     if "last_selected_payer" in session:
         bill_form.payer.data = session["last_selected_payer"]
-    # Preload the "owers" relationship for all bills
-    bills = (
-        g.project.get_bills()
-        .options(orm.subqueryload(Bill.owers))
-        .paginate(per_page=100, error_out=True)
-    )
+    bills = g.project.get_bill_weights_ordered().paginate(per_page=100, error_out=True)
```

Comment:
`bills` is actually a list of tuples here, right? Please rename the variable accordingly :)

Reply:
I hope this is clearer. I didn't go for a very long name like `tuples_of_weights_and_bills`, but decided to add a comment.

However, the template still uses `bills`. I hoped to integrated the weight sum directly into the `Bill` object, it should much much clearer, but… I don't know (yet?) how to do it.

Reply:
In Django, it's done with a function called `annotate`, I don't know if something like this exists in SQLAlchemy

Reply:
After a quick search, `column_property` looks like it could be useful, but I don't understand the documentation... https://docs.sqlalchemy.org/en/14/orm/mapped_sql_expr.html

Simpler way adapted from https://stackoverflow.com/a/14064418 :

```python
def annotate(query):
    for bill, weight in query:
        bill.weight = weight
        yield bill
```

If it works it could even be done directly in `get_bill_weights`.

Reply:
Hi @Glandos, I hope I didn't discourage you with this annotation thing :)

In my opinion it's optional: it would clarify the code to have it, but the most important part of your changes is the performance improvement patch.

Reply:
OK, I wrote this in the comment. Let's not focus on being perfect at first shot.